sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -287,7 +286,7 @@ class Dataset[T] private[sql](
       _numRows: Int,
       truncate: Int = 20,
       vertical: Boolean = false): String = {
-    val numRows = _numRows.max(0).min(Int.MaxValue - 1)
+    val numRows = _numRows.max(0).min(ByteArrayMethods.MAX_ROUNDED_ARRAY_LENGTH - 1)
```

Comment:
Is the "- 1" really necessary after migrating form `Int.MaxValue - 1` to  `ByteArrayMethods.MAX_ROUNDED_ARRAY_LENGTH - 1`?

Reply:
I think it is -- we make a `Seq` potentially one larger than this value here:

https://github.com/apache/spark/blob/361bf02bbe3f78c7a68e93b77fbc9a0ccf39b47a/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L294

(admittedly its a stretch, you shouldn't be showing 2B rows anyway)